Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1124,6 +1125,14 @@ public static void Inline(Program program)
       return outcome;
     }
 
+    private static void ThrowIfCanceled(AggregateException ae) {
+      var canceledException =
+        ae.Flatten().InnerExceptions.FirstOrDefault(exception => exception is OperationCanceledException, null);
+      if (canceledException != null) {
+        throw canceledException;
+      }
```

Comment:
When I have to do something with side-effects on a collection I find that a plain "foreach" works better than Linq:

```
  foreach (var e in ae.Flatten().InnerExceptions) {
    if (e is OperationCanceledException) {
      throw e;
    }
  }
```

Or, if we're trying to keep things as short as possible:

```
  foreach (var e in ae.Flatten().InnerExceptions.OfType<OperationCanceledException>()) {
    throw e;
  }
```